app/actors/curation_concerns/actors/apply_order_actor.rb

```diff
@@ -3,11 +3,28 @@ module Actors
     class ApplyOrderActor < AbstractActor
       def update(attributes)
         ordered_member_ids = attributes.delete(:ordered_member_ids)
+        sync_members(ordered_member_ids)
         apply_order(ordered_member_ids) && next_actor.update(attributes)
       end
 
       private
 
+        def sync_members(ordered_member_ids)
+          return true if ordered_member_ids.nil?
+          existing_members_ids = curation_concern.members.select { |k| k.class.to_s != 'FileSet' }.map(&:id)
```

Comment:
Why not FileSets? They're members too.

Also, `curation-concerns.members` is a really expensive operation. Can you just check the ordered_member_ids?